tests/validation/tests/v3_api/resource/terraform/rke2/worker/join_agent.sh

```diff
@@ -10,12 +10,28 @@ node-name: "${hostname}"
 cloud-provider-name:  "aws"
 EOF
 
+echo "$7"
+if [[ ! -z "$7" ]] && [[ "$7" == *":"* ]]
+then
+   echo "$7"
+   echo -e "$7" >> /etc/rancher/rke2/config.yaml
+   cat /etc/rancher/rke2/config.yaml
+fi
+
 if [ ${1} = "rhel" ]
 then
-    subscription-manager register --auto-attach --username=${6} --password=${7}
-    subscription-manager repos --enable=rhel-7-server-extras-rpms
+   subscription-manager register --auto-attach --username=${8} --password=${9}
+   subscription-manager repos --enable=rhel-7-server-extras-rpms
 fi
 
-curl -sfL https://get.rke2.io | INSTALL_RKE2_VERSION=${5} INSTALL_RKE2_TYPE='agent' sh -
-systemctl enable rke2-agent
-systemctl start rke2-agent
\ No newline at end of file
+if [ ${6} = "rke2" ]
+then
+   curl -sfL https://get.rke2.io | INSTALL_RKE2_VERSION=${5} INSTALL_RKE2_TYPE='agent' sh -
+   sudo systemctl enable rke2-agent
+   sudo systemctl start rke2-agent
+else
+   curl -sfL https://get.rancher.io | INSTALL_RANCHERD_VERSION=${5} INSTALL_RKE2_TYPE='agent' sh -
+   sudo systemctl enable rancherd-agent
+   sudo systemctl start rancherd-agent
+fi
+sleep 180
```

Comment:
I'm wondering if we should separate the "Wait for ready" script from the join_master and install_master into its own shell, and then call it twice: After bringing up node 1 and then again after bringing up the rest of the nodes, whether servers or agents. It would probably save some time and remove the need for the sleep here.